_data-prepper/common-use-cases/pattern-matching.md

```diff
@@ -0,0 +1,215 @@
+---
+layout: default
+title: Grok pattern matching with Data Prepper
```

Comment:
Should we rename this section to be more generic than grok pattern matching. Users have many different scenarios for extracting data. And I think we could eventually make this page guide users through those different scenarios.

I think the content for now is good, but it may be good to establish this page as a broader page.

Perhaps give this a title of "Text processing"

Reply:
@dlvenable Do we need to add any other processors besides `grok` at this time?

Reply:
@vagimeli , We could add `kv` and `dissect` processors now. These provide similar behaviors and we've seen customers use `grok` when they could use `kv` or `dissect`.

I'm not trying to expand the scope of this current PR. I was thinking we could get the naming setup to add these and then have a follow-on PR.

Reply:
@dlvenable I'll open a separate PR to add those processors. I can work on it this week.